header/p2p/options.go

```diff
@@ -0,0 +1,134 @@
+package p2p
+
+import (
+	"fmt"
+	"time"
+)
+
+// Option is the functional option that is applied to the exchange instance
+// to configure parameters.
+type Option[T any] func(*T)
```

Comment:
Instead of this being any, we can define a new type that encompasses both params structs. Look at the Blog and Posts example in https://itnext.io/how-to-use-golang-generics-with-structs-8cabc9353d75
 
This approach provides compile-time type safety

Reply:
done